SWIPs/swip-20-honey_to_money.md

```diff
@@ -0,0 +1,103 @@
+- Zero marginal costs of serving data with perfect competition means that when nodes would engage in peer-to-peer price discovery, the price will tend to zero and a sub-optimal equilibrium of near-zero prices will be reached.
+- The high-quantity of transactions implies that nodes cannot engage in price-discovery for every byte of data transacted, meaning they have to agree *beforehand* on how they will come to a price *at a later time*.
+- Nodes are most of the time just conduits for data and payments, meaning that most of the time data requests (and thus also payments) will take place because nodes want to send data to a peer who requested this data from them.
+
+### Discussion on time differences causing imbalances
+#### Delay between sending and receiving a payment
+Nodes might apply different honey prices for the same payment, due to a time difference between applying a price for *sending* a payment and applying a price for *receiving* the same payment. The imbalances caused by this don't do significant harm if the sum of all imbalances is not greater than the disconnect threshold. Theoretically, if the overall price movement does not show a clear trend (either up or down) over the period where nodes do business with each other, or the two nodes are neither a net receiver nor a net provider to each other, accounting imbalances should offset each other. Furthermore, less frequent as well as small price updates should minimize the imbalances. If a clear trend exists and one node is a net receiver and the other a net provider, the disconnect threshold will eventually be reached. Research must be performed on how the strength of a trend influences the duration until a disconnect is reached for a given stream of traffic and whether this is expected to cause harm to the connectivity in Swarm. The assumption is that this will not be the case for a typical Swarm node. 
+
+If there appears to be a problem with such imbalances, we see a possible solution: as nobody benefits from price imbalances, both the sender and the receiver of a payment are incentivized to apply a price which is not per their clock, but per the clock of the counterparty when they expect the counterparty to query or have queried the oracle. To assist in this, a payload might be sent with the transaction, specifying the clock with which the oracle price was applied.
```

Comment:
> both the sender and the receiver of a payment are incentivized to apply a price which is not per their clock, but per the clock of the counterparty when they expect the counterparty to query or have queried the oracle. 

I don't understand this sentence. If both consult each others' clocks instead of their own, how is the result different than both consulting their own clocks. What dpes the sum of the two "clocks" determine?

Reply:
Please note that this section part is under the discussion sector. If we were to make an implementation based on this, we definitely have to clarify more, but let me give an example of what this *could* mean:

If there is indeed a delay between the time when a receiver of payment queries the oracle and when the sender does so and this delay causes disconnects, we could add an extra field which stipulates at which block the oracle was/will be queried with respect to this transaction.

The counterparty now knows the exact time a conversion happened and can query the blockchain at that time, to get the same conversion. 

Obviously, this query has to be somewhere in between bounds: if a sender of a payment tells he did the conversion 2 years ago, the other node will obviously not apply that conversion and just use the most recent conversion.

That was an example of the receiver applying the conversion "per the clock of the sender". A sender of payment can obviously also predict at what time a receiver will query the oracle, and can signal in the payload that he will defer querying the oracle up until that time. In this case, the sender queries the oracle per the clock of the receiver.

Again, these ideas are not worked out yet, but it might be a possible solution if it happens that imbalances due to a time delay between querying the oracle is a problem.